cmrf_views/src/Plugin/views/query/API4.php

```diff
@@ -0,0 +1,525 @@
+              if ($field_alias = self::getFieldAlias($view->storage->get('base_table'), $key)) {
+                $base_result[$field_alias] = $value;
+              }
+            }
+            $view->result[] = new CMRFViewsResultRow($base_result);
+          }
+          // Set row indices for template_preprocess_views_view_fields to be
+          // able to retrieve the values.
+          array_walk($view->result, function (ResultRow $row, $index) {
+            $row->index = $index;
+          });
+        }
+      }
+
+      // TODO: Adjust relationships for APIv4.
+      foreach ($view->relationship as $field_name => $relationship) {
+        $field_name = self::getFieldAlias($view->storage->get('base_table'), $field_name);
+        $referenced_keys = [];
+        foreach ($view->result as $row_key => &$row) {
+          if (isset($row->{$field_name})) {
+            $referenced_keys[] = $row->{$field_name};
+          }
+        }
+        if (!empty($referenced_keys)) {
+          $base_field_alias = self::getFieldAlias(
+            $relationship->tableAlias,
+            $relationship->getBaseField()
+          );
+          $dataset_relationship = $relationship->getDatasetRelationship();
+          $referenced_dataset = CMRFDataset::load($dataset_relationship->referenced_dataset);
+          // Add Views filters and sorts.
+          $parameters = $this->calculateApiParameters($parameters);
+          // Restrict to foreign keys in current result set.
+          $parameters[$relationship->getBaseField()] = ['IN' => $referenced_keys];
+          // Add dataset parameters, overriding already set values.
+          $parameters = $referenced_dataset->params + $parameters;
+          $options = [
+            'cache' => empty($view->query->options['cache']) ? NULL : $view->query->options['cache'],
+            'limit' => 0,
+          ];
+          $call = $this->core->createCall(
+            $referenced_dataset->connector,
+            $referenced_dataset->entity,
+            $referenced_dataset->action,
+            $parameters,
+            $options,
+            NULL,
+            $referenced_dataset->api_version
+          );
+          $this->core->executeCall($call);
+          if ($call->getStatus() == Call::STATUS_DONE) {
+            $result = $call->getReply();
+            if ((!empty($result['values'])) && (is_array($result['values']))) {
+              foreach ($result['values'] as $relationship_row) {
+                // Filter for needed fields only and rename with aliases.
+                $relationship_result = [];
+                foreach ($relationship_row as $key => $value) {
+                  if ($field_alias = self::getFieldAlias($relationship->tableAlias, $key)) {
+                    $relationship_result[$field_alias] = $value;
+                  }
+                }
+                // Add values to corresponding base rows.
+                foreach ($view->result as $row_key => &$row) {
```

Comment:
`$row_key` is unused. There's no assignment to `$row` so it does not need to be prefixed by `&`.